### PR TITLE
[net] Add trusted unix domain socket implementation

### PR DESCRIPTION
**Pull request by jessehui**

### CODE REVIEW

File: src/libos/crates/async-io/src/util/channel.rs [hunk 1]
```diff
@@ -158,9 +181,83 @@ impl Producer {
...
+        }
+    }
+
+    fn try_writev(&self, bufs: &[&[u8]]) -> Result<usize> {
+        let mut total_nbytes = 0;
+        for buf in bufs {
+            let ret = self.write(buf);
+            if let Ok(nbytes) = ret {
+                total_nbytes += nbytes;
+            }
```
Comment (guzongmin) [inline src/libos/crates/async-io/src/util/channel.rs:249]: Should we return the error here?

File: src/libos/crates/async-io/src/util/channel.rs [hunk 2]
```diff
@@ -262,9 +357,82 @@ impl Consumer {
...
+        }
+    }
+
+    fn try_readv(&self, bufs: &mut [&mut [u8]]) -> Result<usize> {
+        let mut total_nbytes = 0;
+        for buf in bufs {
+            let ret = self.read(buf);
+            if let Ok(nbytes) = ret {
+                total_nbytes += nbytes;
+            }
```
Comment (guzongmin) [inline src/libos/crates/async-io/src/util/channel.rs:424]: return the error

File: src/libos/crates/async-io/src/util/channel.rs (hunk 1 above)
Comment (jessehui) [reply src/libos/crates/async-io/src/util/channel.rs:258]: Refine the logic here. Thanks for your review.

File: src/libos/crates/async-io/src/util/channel.rs (hunk 2 above)
Comment (jessehui) [reply src/libos/crates/async-io/src/util/channel.rs:442]: ditto.

**Comment by jessehui** [comment]
This PR has merged with #135

**Pull request closed without merging**